gensim/models/word2vec.py

```diff
@@ -1251,7 +1251,10 @@ def _minimize_model(self, save_syn1 = False, save_syn1neg = False, save_syn0_loc
             del self.syn1neg
         if hasattr(self, 'syn0_lockf') and not save_syn0_lockf:
             del self.syn0_lockf
-        self.model_trimmed_post_training = True
+        if not save_syn1 or not save_syn1neg or not save_syn0_lockf:
```

Comment:
Do you mean when all 3 are true? Then use `if (save_syn1 and save_syn1neg and save_syn0_lockf)` and just return. Please move this check to the top of the function.